x-pack/plugins/lens/public/datasources/form_based/utils.test.tsx

```diff
@@ -124,7 +124,7 @@ describe('indexpattern_datasource utils', () => {
 
         expect({ ...warningMessages[0], longMessage: '' }).toMatchSnapshot();
 
-        render(<I18nProvider>{warningMessages[0].longMessage}</I18nProvider>);
+        render(<I18nProvider>{warningMessages[0].longMessage as React.ReactNode}</I18nProvider>);
```

Comment:
This is worth a follow-up (or feel free to help in this PR) 

The `longMessage` is typed as `React.ReactNode | ((closePopover: () => void) => React.ReactNode);` but used in the most places simply as a `ReactNode`, which is incorrect

Ideally, this should simply be ` ReactNode` or a `typeof === function` case should be handled with every render.